charts/transfersh/templates/deployment.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: {{ include "transfersh.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "transfersh.name" . }}
+    helm.sh/chart: {{ include "transfersh.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "transfersh.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "transfersh.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      {{- $keyRefSource := .Values.argValues.source.type -}}
+      {{ $keyRefName := .Values.argValues.source.name }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+          {{- range .Values.argValues.paramNames }}
+            - name: {{ . | upper | replace "-" "_" }}
+              valueFrom:
+                {{ $keyRefSource }}:
+                  name: {{ $keyRefName }}
+                  key: {{ . | upper  | replace "-" "_"}}
+          {{- end }}
+          args: [ {{- range .Values.argValues.paramNames }} {{ printf "%s%s" $.Values.argIdentifier . | quote}},{{ printf "%s%s%s" "$(" . ")" | upper | replace "-" "_" | quote}},{{- end }} ]
```

Comment:
for the moment, assuming we will add the missing supported params, it should work, because currently no param is mutually exclusive to each other
tomorrow this would not be true, anyway: we may have conflicting params

also, even if I've not tested, for bool param having the flag set could be enough to enable them: so passing them by default should be avoided, and rather passed or not according to their value

Reply:
Hmm, I guess I understand what you mean.
But isn't the job of params validation & their compatibility should be taken care by the binary itself and not the wrapping frameworks(docker,CI/CD,kubernetes etc.).  
If the params have conflict, shouldn't they be just passed through from outside to the binary and let binary handle them & give appropriate error.

For bool values, you point makes sense. But since our param names are not indicating if they are bool or not, then there must be a way in helm layer to know the difference between a bool param & otherwise.
May be you can give me some hint on some you would want Values.yaml to be(specifically declaring params), so that I can backtrack & understand better.

Reply:
the binary does params validation: if you pass `--provider local` without `--basedir` it won't start
tomorrow it could be the case that if you pass params related to S3 when using local provider will fail validation as well: if the chart just passes everything it won't work or will enable functionality that use doesn't want to enable (like with the bool flags)

I'm not such an expert of helm so I don't know what's the most idiomatic way to solve this issue: I guess having different `values.yaml` for every different provider can be a starting point. for bool flags we could use `flag.enabled` like you did for the ingress and then pass `--set flag.enabled=true` to helm

people that has a stable/immutable deployment needing can write their own custom `values.yaml`, others can rely on the default provided ones and the `--set` flag

Reply:
I get it now, what you mean here & above.  
Give me some time, let me get back to you on how I think what you say, can be implemented (only as a plan) & if we agree, then probably I can fix changes in code.